src/commands/dockersupport/localDockerSupport.ts

```diff
@@ -14,51 +14,43 @@ import { validateDockerInstalled } from "./validateDockerInstalled";
-                            } else {
-                                // Not windows: display link to download docker externally from Docker documentation
-                                void vscode.window.showInformationMessage(localize('downloadDocker', 'Check the (Docker documentation)[https://docs.docker.com/get-docker/] to download Docker for your system'));
-                            }
-                        } else {
-                            void vscode.window.showInformationMessage(localize('noDocker', 'Continuing without the use of Docker as user requested'));
-                        }
+    if (node && devContainerName && node.site.reserved) {
+        // asks if the user wants to use Docker for initializing the project locally
+        const useDocker: string = await prompt(context);
+        if (useDocker === "yes") {
+            await downloadLocalDevFiles(devContainerFolderPathUri, devContainerName);
+            // external - check if Docker is installed, Remote Development extension AND Docker Extension
+            if (!validateDockerInstalled()) {
+                // if Docker is not downloaded - ask the user if they'd like to download Docker
+                if (await prompt2(context) === "yes") {
+                    // Check if Operating System is Windows
+                    if (process.platform == "win32") {
+                        // Download Docker with an MSI package
+                    } else {
+                        // Not windows: display link to download docker externally from Docker documentation
+                        void vscode.window.showInformationMessage(localize('downloadDocker', 'Check the (Docker documentation)[https://docs.docker.com/get-docker/] to download Docker for your system'));
                     }
                 } else {
                     void vscode.window.showInformationMessage(localize('noDocker', 'Continuing without the use of Docker as user requested'));
                 }
-            } else {
-                void vscode.window.showInformationMessage(localize('projectError', 'Function App is not Linux. Continuing without the use of Docker'));
             }
         } else {
-            void vscode.window.showInformationMessage(localize('runtimeError', 'Language not supported. Continuing without the use of Docker'));
+            void vscode.window.showInformationMessage(localize('noDocker', 'Continuing without the use of Docker as user requested'));
         }
     } else {
-        void vscode.window.showInformationMessage(localize('noNode', 'Node is undefined. Continuing without the use of Docker'));
+        void vscode.window.showInformationMessage(localize('unableDocker', 'Initializing project without the use of Docker'));
     }
+}
 
+export async function downloadLocalDevFiles(devContainerFolderPathUri: vscode.Uri, devContainerName?: string): Promise<void> {
+    // download dev container
+    await requestUtils.downloadFile(
```

Comment:
these two calls can be done in parallel. so you could do something like:

```
const downloadDevContainerJson = requestUtils.downloadFile(...);
const downloadDevContainerDockerfile = requestUtils.downloadFile(...);

await Promise.all([
 downloadDevContainerJson,
 downloadDevContainerDockerfile
]);

Reply:
Basically this optimizes your network calls.. lets say if call #1 take 30 seconds and call #2 takes #40 seconds.. the current implementation will take 70 seconds to complete.. but by making the calls parallel, it will take 40 seconds tops.

Reply:
makes sense. I will make this change